payload/src/main/java/io/jenkins/jenkinsfile/runner/TemplatingEnginePipelineDefinitionProvider.java

```diff
@@ -0,0 +1,50 @@
+package io.jenkins.jenkinsfile.runner;
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsUnavailableException;
+import hudson.Extension;
+import io.jenkins.jenkinsfile.runner.bootstrap.commands.PipelineRunOptions;
+import org.apache.commons.io.FileUtils;
+import org.boozallen.plugins.jte.job.*;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension public class TemplatingEnginePipelineDefinitionProvider extends PipelineDefinitionProvider {
+
+    private static final Logger LOGGER = Logger.getLogger(TemplatingEnginePipelineDefinitionProvider.class.getName());
+
+    static final String definition = "Jenkins Templating Engine";
+
```

Comment:
Needs static initializer which checks availability of the class within the plugin.
Otherwise the optional extension will be accepted